docs/source/guides/server.rst

```diff
@@ -2,23 +2,68 @@
 Customize BentoServer
 =====================
 
+BentoServer architecture runs the BentoML Service API in an `ASGI <https://asgi.readthedocs.io/en/latest/>`_
+web serving layer and Runners in a separate work process pool manged by BentoML. The ASGI web
```

Comment:
For the recurring term of "web serving layer", should we just call it "web server" or "http server"?

Reply:
We can discuss separately. No reason to hold on to this PR.